libraries/chain/include/steem/chain/smt_objects/smt_token_object.hpp

```diff
@@ -206,6 +182,7 @@ class smt_contribution_object : public object< smt_contribution_object_type, smt
 };
 
 struct by_symbol_contributor;
+struct by_contributor;
 
 typedef multi_index_container <
```

Comment:
We need to do a little thinking ahead here. To complete the ICO we need to be able to iterate through contributions in the order in which they came in. The contribution id is important from a UI perspective, but doesn't actually give us what we need.

Thankfully, we can leverage the fact that IDs are monotonically increasing. So if we create a composite key on `(symbol,id)`, you will have an in order index of contributions for an SMT.